internal/driver/webui.go

```diff
@@ -131,6 +123,34 @@ func serveWebInterface(hostport string, p *profile.Profile, o *plugin.Options, w
 	return server(args)
 }
 
+func getHostAndPort(hostport string) (string, int, error) {
+	host, portStr, err := net.SplitHostPort(hostport)
+	if err != nil {
+		return "", 0, fmt.Errorf("could not split http address: %v", err)
+	}
+	if host == "" {
+		host = "localhost"
+	}
+	var port int
+	if portStr == "" {
+		ln, err := net.Listen("tcp", net.JoinHostPort(host, "0"))
+		if err != nil {
+			return "", 0, fmt.Errorf("could not generate random port: %v", err)
+		}
+		port = ln.Addr().(*net.TCPAddr).Port
+		err = ln.Close()
```

Comment:
I think this is racy in the sense that another program may grab the port after the socket is closed here and opened again by the HTTP server. Did you consider this possibility? Is this too unlikely to worry about?

Reply:
It's definitely possible -- I believe it will occur rarely and I could not find another way to generate an unused port. 

To reduce this possibility, I could shift the functionality of getHostAndPort() into serveWebInterface(), and close the connection later. But that would make it harder to test random port selection and still not completely eliminate the race.